annotated.go

```diff
@@ -312,6 +316,33 @@ func (ann *annotated) Build() (interface{}, error) {
 	return newFn.Interface(), nil
 }
 
+// checks whether the target function is either
+// returning an fx.Out struct or an taking in a
+// fx.In struct as a parameter.
+func (ann *annotated) typeCheckOrigFn() error {
+	ft := reflect.TypeOf(ann.Target)
+	for i := 0; i < ft.NumOut(); i++ {
+		ot := ft.Out(i)
+		if ot.Kind() != reflect.Struct {
+			continue
+		}
+		if ot.NumField() > 0 && ot.Field(0).Type == reflect.TypeOf(Out{}) {
```

Comment:
Are we assuming the first Field is always `Out{}` ?

Reply:
I swear I remember reading somewhere that In/Out embeddings should be the first field in a struct, but I am unable to find any supporting docs now. Maybe @sywhang can find some. 🤷‍♀️

Reply:
There's no such restriction. The `fx.Out` can be any field.

I haven't reviewed the PR yet, but I think Sung is validating the generated function, in which the `fx.Out` will indeed be the first field.

Reply:
oops. I'm validating the input function, so I think this should be fixed. lemme follow up with a fix